torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -914,19 +916,134 @@ void PeepholeOptimizeShapeExpressions(Block* block) {
   }
 }
 
+// Note: shamelessly copying code from NNC
```

Comment:
this is a useful note, but can you please expand (what is NCC, and maybe add a link/path to the original code)?